editor.planx.uk/src/@planx/graph/index.ts

```diff
@@ -44,12 +45,14 @@ const isSectionNodeType = (id: string, graph: Graph): boolean =>
 const isExternalPortalNodeType = (id: string, graph: Graph): boolean =>
   graph[id]?.type === TYPES.ExternalPortal;
 
-const sanitize = (x: any) => {
+export const sanitize = (x: any) => {
   if ((x && typeof x === "string") || x instanceof String) {
+    x = DOMPurify.sanitize(x as string);
     return trim(x.replace(/[\u200B-\u200D\uFEFF↵]/g, ""));
   } else if ((x && typeof x === "object") || x instanceof Object) {
     return Object.entries(x).reduce((acc, [k, v]) => {
       v = sanitize(v);
+      acc[k] = v;
```

Comment:
This block only `returns acc`, and `v = sanitize(v)` was indeed sanitizing, but not actually updating the `acc` which would be returned: 
- In cases of handling empty values, this didn't matter because we `delete acc[k]`
- But in cases of sanitising values we want to keep, we need to update the return value too